internal/pkg/crawl/extractor/xml.go

```diff
@@ -1,14 +1,18 @@
 package extractor
 
 import (
+	"encoding/xml"
 	"io"
 	"net/http"
 	"net/url"
 	"strings"
-
-	"github.com/clbanning/mxj/v2"
 )
 
+type LeafNode struct {
+	Path  string `json:"path"`
+	Value string `json:"value"`
+}
+
```

Comment:
Hmm, is LeafNode not involved in the actual code logic?

Reply:
Isn't it? I make mark of the current element and an array of leafnodes using this struct

Reply:
![image](https://github.com/user-attachments/assets/8e3bf617-6264-4931-aab8-0c861b7ee40f)

To me, they are useless codes. After all, removing them doesn't affect the function, right?

Reply:
I aimed for a full xml parsing logic, thinking it maybe needed in the future. Didn't aim for only getting the urls, but the full markup.

Just let me check and I will remove extra code

Reply:
All Right, all tests pass if I remove the element tracking logic and leafNodes tracker. LeafNode is not required right now. I will push